src/libraries/Microsoft.Extensions.Options/src/OptionsSnapshot.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Implementation of <see cref="IOptions{TOptions}"/> and <see cref="IOptionsSnapshot{TOptions}"/>.
+    /// </summary>
+    /// <typeparam name="TOptions">Options type.</typeparam>
+    internal class OptionsSnapshot<[DynamicallyAccessedMembers(Options.DynamicallyAccessedMembers)] TOptions> :
+        IOptionsSnapshot<TOptions>
+        where TOptions : class
+    {
+        private readonly IOptionsMonitor<TOptions> _factory;
```

Comment:
Calling this `factory` is a bit odd, IMO.

Reply:
Can rename it no problem, though the monitor serves the function of a factory or provider here IMO.